.circleci/config.yml

```diff
@@ -0,0 +1,155 @@
+    resource_class: << parameters.resource_class >>
+    working_directory: ~/project
+    steps:
+      - checkout
+      - restore_cache:
+          key: cegcc-{{ arch }}-{{ checksum "Dockerfile.worker" }}-{{ checksum "Makefile" }}-{{ checksum "scripts/build_cegcc.bash" }}
+      - run: 
+          command: |
+            export ARCH=$(make show-host-arch)
+            if [ ! -e cache/cegcc_${ARCH}.tar.xz ]; then
+              make build-worker
+            else 
+              echo "cache found. skip build process"
+            fi
+      - run:
+          name: build cegcc
+          command: |
+            export ARCH=$(make show-host-arch)
+            if [ ! -e cache/cegcc_${ARCH}.tar.xz ]; then
+              make cache/cegcc_${ARCH}.tar.xz
+            else
+              echo "cache found. skip build process"
+            fi
+      - save_cache:
+          key: cegcc-{{ arch }}-{{ checksum "Dockerfile.worker" }}-{{ checksum "Makefile" }}-{{ checksum "scripts/build_cegcc.bash" }}
+          paths:
+            - cache
+      - persist_to_workspace:
+          root: ~/project
+          paths:
+            - cache
+
+  build_arm-none-eabi:
+    machine:
+      image:  ubuntu-2004:current
+    parameters:
+      resource_class:
+        type: string
+    resource_class: << parameters.resource_class >>
+    working_directory: ~/project
+    steps:
+      - checkout
+      - restore_cache:
+          key: arm-none-eabi-{{ arch }}-{{ checksum "Dockerfile.worker" }}-{{ checksum "Makefile" }}-{{ checksum "scripts/build_arm-none-eabi.bash" }}
+      - run: 
+          command: |
+            export ARCH=$(make show-host-arch)
+            if [ ! -e cache/arm-none-eabi_${ARCH}.tar.xz ]; then
+              make build-worker
+            else 
+                echo "cache found. skip build process"
+            fi
+      - run:
+          name: build arm-none-eabi
+          command: |
+            export ARCH=$(make show-host-arch)
+            if [ ! -e cache/arm-none-eabi_${ARCH}.tar.xz ]; then
+              make cache/arm-none-eabi_${ARCH}.tar.xz
+            else
+              echo "cache found. skip build process"
+            fi
+      - save_cache:
+          key: arm-none-eabi-{{ arch }}-{{ checksum "Dockerfile.worker" }}-{{ checksum "Makefile" }}-{{ checksum "scripts/build_arm-none-eabi.bash" }}
+          paths:
+            - cache
+      - persist_to_workspace:
+          root: ~/project
+          paths:
+            - cache/
+
+  build_and_test_builder:
+    machine:
+      image:  ubuntu-2004:current
+    parameters:
+      resource_class:
+        type: string
+    resource_class: << parameters.resource_class >>
+    steps:
+      - checkout
+      - attach_workspace:
+          at: .
+      - run:
+          command: |
+            export ARCH=$(make show-host-arch)
+            docker buildx build \
+              --load \
+              --platform linux/${ARCH} \
+              --tag brainhackers/gen1_builder:latest .
```

Comment:
(Must) Please replace `brainhackers` with `$DOCKER_USER`. This applies to L97, L101, L105, and L128.